src/box/tuple.c

```diff
@@ -46,6 +46,38 @@ enum {
 	OBJSIZE_MIN = 16,
 };
 
+/**
+ * Storage for additional reference counter of a tuple.
+ */
+struct tuple_uploaded_refs {
+	struct tuple *tuple;
+	uint32_t refs;
+};
+
+static uint32_t
+tuple_pointer_hash(const struct tuple *a)
+{
+	uintptr_t u = (uintptr_t)a;
+	if (sizeof(uintptr_t) <= sizeof(uint32_t))
+		return u;
+	else
+		return u ^ (u >> 32);
+}
+
+#define mh_name _uploaded_refs
+#define mh_key_t struct tuple *
+#define mh_node_t struct tuple_uploaded_refs
+#define mh_arg_t int
+#define mh_hash(a, arg) (tuple_pointer_hash((a)->tuple))
+#define mh_hash_key(a, arg) (tuple_pointer_hash(a))
+#define mh_cmp(a, b, arg) ((a)->tuple != (b)->tuple)
+#define mh_cmp_key(a, b, arg) ((a) != (b)->tuple)
+#define MH_SOURCE 1
+#define MH_INCREMENTAL_RESIZE 1
```

Comment:
Redundant.

Reply:
I thought it would be more clear but ok.